main.gd

```diff
@@ -22,11 +22,23 @@ var responseHeaders = null
 var responseStatus = null
 var jsonParseError = null
 var gameJoltErrorMessage = null
-
+var lasttype=""
+func init(pk,gi):
+	private_key=pk
+	game_id=gi
 func _ready():
 	connect("request_completed", self, '_on_HTTPRequest_request_completed')
 	pass
-	
+func auto_auth():
+	#get username and token form url on gamejolt (only work with html5)
+	var url=str(JavaScript.eval("window.location.href"))
```

Comment:
I rather propose to do something like this:
		username_cache = JavaScript.eval('var urlParams = new URLSearchParams(window.location.search); urlParams.get("username_cache")', true)